RELEASING.md

```diff
@@ -36,49 +36,31 @@ to view a summary of all commits since last release as a reference.
 All patch releases should include only bug-fixes, and must avoid
 adding/modifying the public APIs. 
 
-Open the patch release build workflow in your browser [here](https://github.com/aws-observability/aws-otel-java-instrumentation/actions?query=workflow%3A%22Patch+Release+Build%22).
+Steps:
+1. Create a branch from the release that you want to patch. It should follow the convention `release/v<major>.<minor>.x`. E.g.: if you want to patch release 1.21.0, the name of the branch should be `release/v1.21.x`.
+1. Mark the branch as protected.
```

Comment:
nit: can we use branch protection settings with a wildcard to automatically protect these branches? @Aneurysm9

Reply:
Done.  Let me know if there are any issues.  Team members should have the ability to push new branches matching the pattern `release/v*` and the branch will be automatically protected.